e2e/playwright/testing-samples-loading.spec.ts

```diff
@@ -0,0 +1,183 @@
+
+    await test.step(`Precondition: check the initial code`, async () => {
+      await u.openKclCodePanel()
+      await expect(codeLocator).toContainText(bracket.split('\n')[0])
+    })
+
+    await test.step(`Load a KCL sample with the command palette`, async () => {
+      await commandBarButton.click()
+      await commandOption.click()
+      await commandSampleOption.click()
+      await commandMethodArgButton.click()
+      await expect(commandMethodOption('Create new file')).not.toBeVisible()
+      await commandMethodOption('Overwrite').click()
+      await expect(warningText).toBeVisible()
+      await confirmButton.click()
+
+      await expect(codeLocator).toContainText('// ' + newSample.title)
+    })
+  })
+
+  /**
+   * Note this test implicitly depends on the KCL samples:
+   * "flange-with-patterns.kcl": https://github.com/KittyCAD/kcl-samples/blob/main/flange-with-patterns/flange-with-patterns.kcl
+   * "gear-rack.kcl": https://github.com/KittyCAD/kcl-samples/blob/main/gear-rack/gear-rack.kcl
+   */
+  test(
+    'Desktop: should create new file by default, optionally overwrite',
+    { tag: '@electron' },
+    async ({ browserName: _ }, testInfo) => {
+      const { electronApp, page } = await setupElectron({
+        testInfo,
+        folderSetupFn: async (dir) => {
+          const bracketDir = join(dir, 'bracket')
+          await fsp.mkdir(bracketDir, { recursive: true })
+          await fsp.writeFile(join(bracketDir, 'main.kcl'), bracket, {
+            encoding: 'utf-8',
+          })
+        },
+      })
+      const u = await getUtils(page)
+
+      // Locators and constants
+      const sampleOne = {
+        file: 'flange-with-patterns' + FILE_EXT,
+        title: 'Flange',
+      }
+      const sampleTwo = {
+        file: 'gear-rack' + FILE_EXT,
+        title: '100mm Gear Rack',
+      }
+      const projectCard = page.getByRole('link', { name: 'bracket' })
+      const commandBarButton = page.getByRole('button', { name: 'Commands' })
+      const commandOption = page.getByRole('option', { name: 'Open Sample' })
+      const commandSampleOption = (name: string) =>
+        page.getByRole('option', {
+          name,
+          exact: true,
+        })
+      const commandMethodArgButton = page.getByRole('button', {
+        name: 'Method',
+      })
+      const commandMethodOption = page.getByRole('option', {
+        name: 'Overwrite',
+      })
+      const newFileWarning = page.getByText(
+        'Create a new file with the example code?'
+      )
+      const overwriteWarning = page.getByText('Overwrite current file?')
+      const confirmButton = page.getByRole('button', { name: 'Submit command' })
+      const projectMenuButton = page.getByTestId('project-sidebar-toggle')
+      const newFile = (name: string) =>
+        page.getByRole('listitem').filter({
+          has: page.getByRole('button', { name }),
+        })
+      const codeLocator = page.locator('.cm-content')
```

Comment:
why's this util called `newFile`, it's just filtering files in the file tree by a name right. Reading it's usage below, I thought maybe it was creating something.

Reply:
Ah gonna change it to `newlyCreatedFile`